vdr/ambassador.go

```diff
@@ -134,16 +144,45 @@ func (n *ambassador) handleCreateDIDDocument(transaction dag.Transaction, propos
 		return err
 	}
 
+	// check for an existing document, it could have been a parallel create. If existing merge and update.
+	currentDIDDocument, currentDIDMeta, err := n.didStore.Resolve(proposedDIDDocument.ID, nil)
+	if err != nil && !errors.Is(err, types.ErrNotFound) {
+		return fmt.Errorf("unable to create DID document: %w", err)
```

Comment:
not sure whether this is the appropriate message (since you're not creating a document here, just storing a created document in VDR)

Reply:
already was so, but will change to register